llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -3278,7 +3268,10 @@ NVPTXTargetLowering::LowerSTOREVector(SDValue Op, SelectionDAG &DAG) const {
   Ops.push_back(N->getOperand(0));
 
   // Then the split values
-  if (UpsizeElementTypes) {
+  if (ValVT.getVectorNumElements() > NumElts) {
+    // If the number of elements has changed, getVectorLoweringShape has upsized
+    // the element types
+    assert((Isv2x16VT(EltVT) || EltVT == MVT::v4i8) && "Unexpected upsized type.");
```

Comment:
I think what we really care about here is that EltVT is a vector, is 32 bits in size, and has no more than 4 elements. Checking for specific types does the job, but relies on the implementation detail and we have to change it every time we add another type.

Reply:
Done, changed assert. Can you double check my API usage to make sure I'm representing these checks correctly?